crates/biome_js_type_info/src/inference.rs

```diff
@@ -173,6 +145,125 @@ impl Type {
+                    parameters: match expr.parameters() {
+                        Ok(AnyJsArrowFunctionParameters::AnyJsBinding(binding)) => {
+                            Box::new([FunctionParameter {
+                                name: binding
+                                    .as_js_identifier_binding()
+                                    .and_then(|binding| text_from_token(binding.name_token())),
+                                ty: Type::Unknown,
+                                is_optional: false,
+                                is_rest: false,
+                            }])
+                        }
+                        Ok(AnyJsArrowFunctionParameters::JsParameters(params)) => {
+                            function_params_from_js_params(Ok(params))
+                        }
+                        Err(_) => Box::default(),
+                    },
+                    return_type: return_type_from_annotation(expr.return_type_annotation())
+                        .unwrap_or_else(|| return_type_from_async_token(expr.async_token())),
+                }))
+            }
+            AnyJsExpression::JsFunctionExpression(expr) => Type::Function(Box::new(Function {
+                is_async: expr.async_token().is_some(),
+                type_parameters: generic_params_from_ts_type_params(expr.type_parameters()),
+                name: expr
+                    .id()
+                    .as_ref()
+                    .and_then(|binding| binding.as_js_identifier_binding())
+                    .and_then(|binding| text_from_token(binding.name_token())),
+                parameters: function_params_from_js_params(expr.parameters()),
+                return_type: return_type_from_annotation(expr.return_type_annotation())
+                    .unwrap_or_else(|| return_type_from_async_token(expr.async_token())),
+            })),
+            _ => {
+                // TODO: Much
+                Type::Unknown
+            }
+        }
+    }
+
+    pub fn from_any_js_literal_expression(expr: &AnyJsLiteralExpression) -> Option<Self> {
+        let literal = match expr {
+            AnyJsLiteralExpression::JsBigintLiteralExpression(expr) => {
+                Literal::BigInt(text_from_token(expr.value_token())?)
+            }
+            AnyJsLiteralExpression::JsBooleanLiteralExpression(expr) => {
+                Literal::Boolean(text_from_token(expr.value_token())?)
+            }
+            AnyJsLiteralExpression::JsNullLiteralExpression(_) => Literal::Null,
+            AnyJsLiteralExpression::JsNumberLiteralExpression(expr) => {
+                Literal::Number(text_from_token(expr.value_token())?)
+            }
+            AnyJsLiteralExpression::JsRegexLiteralExpression(expr) => {
+                Literal::RegExp(text_from_token(expr.value_token())?)
+            }
+            AnyJsLiteralExpression::JsStringLiteralExpression(expr) => {
+                Literal::String(text_from_token(expr.value_token())?)
```

Comment:
String literals carry their original quotes, shouldn't we strip them by using `inner_string_text`?

Reply:
Good question. By the time we want to start reasoning about them, we'll probably only care about their contents indeed, in which case we should also unescape them. That's a bit more expensive though, so for now this'll do.